TranslatorReasonerAPI.yaml

```diff
@@ -470,46 +470,75 @@ components:
-        a source of this attribute, and (optionally) a url with additional
-        information about this attribute.
+        Generic attribute for a node or an edge that expands the key-value
+        pair concept by including fields for additional metadata. These fields
+        can be used to describe the source of the statement made in key-value
+        pair of the attribute object, or describe the attribute's value itself
+        including its semantic type, or a url providing additional information
+        about it.
       properties:
-        name:
+        attribute_type_id:
+          $ref: '#/components/schemas/CURIE'
+          description: >-
+            The 'key' of the attribute object, holding a CURIE of an ontology
+            property defining the attribute (preferably the CURIE of a
+            Biolink association slot). This property captures the relationship
+            asserted to hold between the value of the attribute, and the node
+            or edge from  which it hangs. For example, that a value of
+            '0.000153' represents a p-value supporting an edge, or that
+            a value of 'ChEMBL' represents the original source of the knowledge
+            expressed in the edge.
+          example: Biolink:has_p-value_evidence, Biolink:has_original_source
+        original_attribute_name:
           type: string
           description: >-
-            Human-readable name or label for the attribute. If appropriate,
-            should be the name of the semantic type term.
-          example: PubMed Identifier
+            The term used by the original source of an attribute to describe
+            the meaning or significance of the value it captures. This may be
+            a column name in a source tsv file, or a key in a source json
+            document for the field in the data that held the attribute's
+            value. Capturing this information  where possible lets us preserve
+            what the original source said. Note that the data type is string'
+            but the contents of the field could also be a CURIE of a third
+            party ontology term.
+          example: p-value
           nullable: true
         value:
-          example: 32529952
           description: >-
             Value of the attribute. May be any data type, including a list.
-        type:
+          example: 0.000153
+        value_type_id:
           $ref: '#/components/schemas/CURIE'
           description: >-
-            CURIE of the semantic type of the attribute. For properties
-            defined by the Biolink model this should be a biolink CURIE,
-            otherwise, if possible, from the EDAM ontology. If a suitable
-            identifier does not exist, enter a descriptive phrase here and
-            submit the new type for consideration by the appropriate
-            authority.
+            CURIE describing the semantic type of an  attribute's value. Use
```

Comment:
Is the `value_type_id` always, sometimes, or never implied by the `attribute_type_id`? Maybe some examples of the two used together would be helpful.

Reply:
The attribute_type_id seems to describe the semantic association between the value and the object that contains that value, whereas the value_type_id describes only the value itself. I think the similarity would depend on the specificity of the attribute_type_id. Something specific like has_p_value_evidence will probably only ever link to a p-value-related term. Whereas something like has_original_source could be a publication, an assay, another knowledge source, or something else.

Reply:
This is exactly right Eugene.  The idea here is to allow for very granular/specific properties that can be tied to a specific value type. We are exploring if/when this is practical and useful - using the examples in the attribute_types inventory.  If we end up supporting properties specific to each value type, then this field becomes redundant (in that the value type can always be looked up in the Biolink Model).  

But  it may be that there is value in duplicating this info in the message itself to avoid the need to lookup form Biolink .  And we anticipate that there may be cases where a KP wants a value_type that is more precise than we want in Biolink (e.g. indicating that a curie for a Publication represents a DOI vs PMID).  We will assess this as we work through real use cases/examples.